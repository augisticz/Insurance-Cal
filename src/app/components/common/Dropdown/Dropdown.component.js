import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './Dropdown.style.css';
import noop from 'lodash/noop';

class Dropdown extends Component {
  state = {
    selectValue: ''
  };

  handleChange = (e) => {
    this.setState({selectValue: e.target.value});
    this.props.onChange(e.target.value);
  };
    
  render () {
    return (this.props.listData && this.props.listData.length > 0) ? <select className='form-select selection' value={this.state.selectValue} onChange={this.handleChange} >
      <option value=''>กรุณาเลือก</option>
      {this.props.listData.map(({title, value}, i) => <option key={i} value={value}>{title}</option>)}
    </select> : null;
  }
}

Dropdown.defaultProps = {
  listData: [],
  onChange: noop
};
  
Dropdown.propTypes = {
  listData: PropTypes.array,
  onChange: PropTypes.func
};

export default Dropdown;