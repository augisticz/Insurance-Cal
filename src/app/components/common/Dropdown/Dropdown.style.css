.selection {
    margin-left: -13px;
    width: 106%;
}