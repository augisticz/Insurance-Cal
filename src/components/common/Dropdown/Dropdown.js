import { useState } from "react";
import './Dropdown.css';

const Dropdown = (props) => {
    const [selectValue, setSelectValue] = useState('')
    const handleChange = (e) => {
        setSelectValue(e.target.value)
        props.onChange(e.target.value)
    }
    
    return (props.listData && props.listData.length > 0) ? <select className="form-select selection" value={selectValue} onChange={handleChange} >
            <option value="">กรุณาเลือก</option>
            {props.listData.map(({title, value}, i) => <option key={i} value={value}>{title}</option>)}
          </select> : null
}

export default Dropdown;