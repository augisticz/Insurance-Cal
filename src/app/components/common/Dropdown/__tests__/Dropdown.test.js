import Dropdown from '../Dropdown.component';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {genderList} from '../../../../configs/configStaticValue.config';

describe('Dropdown Component', () => {
  const wrapper = shallow(<Dropdown listData={genderList} />);

  it('renders correctly', () => {
    const component = renderer.create(<Dropdown listData={genderList} />).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });

  it('should handleChange', () => {
    const selectValue = {
      target: {
        value: '1'
      }
    };
    wrapper.instance().handleChange(selectValue);
    expect(wrapper.instance().state.selectValue).toEqual('1');
  });
});