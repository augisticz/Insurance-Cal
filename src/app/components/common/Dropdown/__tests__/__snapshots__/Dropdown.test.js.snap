// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component renders correctly 1`] = `
<select
  className="form-select selection"
  onChange={[Function]}
  value=""
>
  <option
    value=""
  >
    กรุณาเลือก
  </option>
  <option
    value="MALE"
  >
    ชาย
  </option>
  <option
    value="FEMALE"
  >
    หญิง
  </option>
</select>
`;
